smarts/core/sumo_traffic_simulation.py

```diff
@@ -240,20 +242,24 @@ def _base_sumo_load_params(self):
 
         return load_params
 
-    def setup(self, scenario) -> ProviderState:
+    def setup(self, next_scenario) -> ProviderState:
         self._log.debug("Setting up SumoTrafficSim %s" % self)
         assert not self._is_setup, (
             "Can't setup twice, %s, see teardown()" % self._is_setup
         )
 
         # restart sumo process only when map file changes
-        if self._scenario and self._scenario.net_file_hash == scenario.net_file_hash:
-            restart_sumo = False
-        else:
-            restart_sumo = True
+        restart_sumo = (
+            not (
+                self._scenario
+                and self._scenario.net_file_hash == next_scenario.net_file_hash
+            )
+            or self._current_reload_count >= self._reload_count
+        )
+        self._current_reload_count = self._current_reload_count % self._reload_count + 1
```

Comment:
If you only reload SUMO there will be a gradual and uncapped memory growth. 

I will need to change the comment here.

Reply:
This seems ok to me, although I have a slight preference to "flatten" the boolean logic, like:
```python
restart_sumo = (
    not self._scenario
    or self._scenario.net_file_hash != next_scenario.net_file_hash
    or self._current_reload_count >= self._reload_count
)
```

If there isn't a Sumo option for resetting, we should probably file an Issue with them letting  them know they might have a leak.